File: tests/fixtures/dart/classes.dart
class Calculator {
  int value = 0;

  int add(int a, int b) {
    value = a + b;
    return value;
  }

  void print(String message) {
    print(message);
  }
}

class Point {
  double x;
  double y;

  Point(this.x, this.y);
}

interface IProcessor {
  void process(String data);
}

enum Color {
  red,
  green,
  blue
}

mixin Printable {
  void printMessage();
}

extension StringExtensions on String {
  String trimmed() {
    return trim();
  }
}

File: tests/fixtures/swift/classes.swift
import Foundation

class Calculator {
    private var value: Int = 0

    func add(a: Int, b: Int) -> Int {
        value = a + b
        return value
    }

    func print(message: String) {
        print(message)
    }
}

struct Point {
    let x: Double
    let y: Double
}

protocol IProcessor {
    func process(data: String)
}

enum Color {
    case red
    case green
    case blue
}

typealias StringMap = Dictionary<String, String>

extension String {
    func trimmed() -> String {
        return self.trimmingCharacters(in: .whitespaces)
    }
}
